tools/toolchains/arm.py

```diff
@@ -594,3 +640,21 @@ def compile(self, cc, source, object, includes):
         cmd.extend(self.get_compile_options(self.get_symbols(), includes))
         cmd.extend(["-o", object, source])
         return [cmd]
+
+    def get_link_command(self, output, objects, libraries, lib_dirs, scatter_file):
+        cmd = ARM.get_link_command(
+            self, output, objects, libraries, lib_dirs, scatter_file
+        )
+
+        if self.is_mbed_studio_armc6:
+            cmd.insert(1, "--ide=mbed")
```

Comment:
IMO, this flag should be passed in whenever running AC6, not specifically when `is_mbed_studio_armc6`. The detection of Mbed Studio version can then be removed.

Reply:
It was deemed necessary to do the conditional insertion of the `--ide=mbed` flag because supplying this to the professional version of ARMC6 causes a warning to show for every compile, assemble, and link step. The warning is along the lines of "unused argument --ide=mbed".